sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -314,7 +323,7 @@ static ISymbol GetOriginalDefinition(ISymbol candidateSymbol)
 
         private void TryStorePropertyAccess(ExpressionSyntax node, List<ISymbol> identifierSymbols)
         {
-            var propertySymbols = identifierSymbols.OfType<IPropertySymbol>();
+            var propertySymbols = identifierSymbols.OfType<IPropertySymbol>().ToList();
```

Comment:
why is the ToList necessary? because of the Any() followed by the foreach?

Reply:
exactly. the enumeration is evaluated twice